lib/puppetserver/utils/http_utilities.rb

```diff
@@ -74,11 +75,24 @@ def to_uri
               end
             end
 
-      def self.make_store(cacert, cert, crl)
+      def self.make_store(bundle, crl_usage, crls = nil)
         store = OpenSSL::X509::Store.new
-        store.add_file(cacert)
-        store.add_file(cert)
-        store.add_file(crl)
+        store.purpose = OpenSSL::X509::PURPOSE_ANY
+        store.add_file(bundle)
+
+        if crl_usage != :ignore
+
+          flags = OpenSSL::X509::V_FLAG_CRL_CHECK
+          if crl_usage == :chain
+            flags |= OpenSSL::X509::V_FLAG_CRL_CHECK_ALL
+          end
+
+          store.flags = flags
+          delimiter = /-----BEGIN X509 CRL-----.*?-----END X509 CRL-----/m
+          File.read(crls).scan(delimiter).each do |crl|
+            store.add_crl(OpenSSL::X509::CRL.new(crl))
```

Comment:
I'm assuming the PKI for the agent has been set up correctly already, and we've got a lot of logic in the agent for that kind of thing.

I'm assuming we should have some kind of rescue around this, but that we can get away with a "Woah, stuff's broke, go fix your agent's PKI first" kinda thing?

Reply:
Yeah I think so.

Reply:
I added something like this when debugging the issue on a VM:
```rb
        begin
          store = make_store(settings[:localcacert],
                             settings[:certificate_revocation],
                             settings[:hostcrl])
          request = ->(conn) { block.call(Connection.new(conn, url)) }
          result = Net::HTTP.start(url.host, url.port,
                          use_ssl: true, cert_store: store,
                          &request)
          return result
        rescue => e
          STDOUT.puts 'Unable to complete request!'
          STDOUT.puts 'Attempting connection to URL:'
          STDOUT.puts pp(url)
          STDOUT.puts 'With SSL Store:'
          STDOUT.puts pp(store)
          STDOUT.puts 'Encountered exception:'
          STDOUT.puts pp(e)
          raise e
        end
```

And I realized that we should pass in a logger here if we're going to do this correctly. What do you think about doing that when we turn it into a class/client?